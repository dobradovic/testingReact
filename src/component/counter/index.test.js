import React from "react";
import { shallow } from "enzyme";
import Counter from "./index";
import { findByTestAttr } from "./../../../Utils";

describe("Testing counter", () => {
  it("should reneder component correctly", () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper).toMatchSnapshot();
  });

  it("the increment method increments count", () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper.instance().state.count).toBe(0);
    // wrapper.find('button.counter-button').simulate('click')
    // wrapper.setState({count: 1})
    wrapper.instance().increment();
    expect(wrapper.instance().state.count).toBe(1);
  });
});
