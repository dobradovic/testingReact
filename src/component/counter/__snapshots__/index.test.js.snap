// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing counter should reneder component correctly 1`] = `ShallowWrapper {}`;
